osu.Game/OsuGame.cs

```diff
@@ -1023,9 +1010,14 @@ protected override void UpdateAfterChildren()
         {
             base.UpdateAfterChildren();
 
-            screenOffsetContainer.Padding = new MarginPadding { Top = ToolbarOffset };
+            ScreenOffsetContainer.Padding = new MarginPadding { Top = ToolbarOffset };
             overlayContent.Padding = new MarginPadding { Top = ToolbarOffset };
 
+            ScreenOffsetContainer.X = 0f;
+
+            if (Settings.IsLoaded) ScreenOffsetContainer.X += (ToLocalSpace(Settings.ScreenSpaceDrawQuad.TopRight).X) * SCREEN_OFFSET_RATIO;
+            if (Notifications.IsLoaded) ScreenOffsetContainer.X += (ToLocalSpace(Notifications.ScreenSpaceDrawQuad.TopLeft).X - DrawWidth) * SCREEN_OFFSET_RATIO;
```

Comment:
These every-frame operations can probaby be avoided when the overlay is not visible.

Reply:
Actually now I remember why I haven't put that, using `State.Value == Hidden` would make the offset transition become instant rather than being smoothed out, as the state change usually happens before the transforms apply.

Seems like this one cannot be easily solved, but I think it's not much of an expensive operation either way. (although I'm just assuming at this point)

Reply:
why can't you check `IsPresent`? aka visibility.

Reply:
right